### ISSUE TITLE
SPHttpClient / GraphHttpClient constructor parameter is no longer optional

### ISSUE THREAD

**Issue by marwankhalili** [pnp/pnpjs#1620]
### Category
- [x] Bug

### Version

Please specify what version of the library you are using: [2.2.0]

### Expected / Desired Behavior / Question
The SPHttpClient should not require a parameter for the constructor. I assume this issue applies to GraphHttpClient as well but I have not tested it.

### Observed Behavior
This issue was introduced in v2.1.0 from changes in this commit c240dfe91fe3a76533ddfda65ade14fb768361a9. 
```typescript
constructor(...args: [runtime: Runtime] | [impl: IHttpClientImpl, runtime?: Runtime])
```
Not sure what solution is most elegant. Some suggestions that might work:
```typescript
constructor(...args: [runtime?: Runtime] | [impl: IHttpClientImpl, runtime?: Runtime])
constructor(...args: [] | [runtime: Runtime] | [impl: IHttpClientImpl, runtime?: Runtime])
```

### Steps to Reproduce
- `npm i @pnp/sp@2.2.0`
- `npm i --save-dev typescript`
-  `echo 'import {SPHttpClient} from "@pnp/sp"; new SPHttpClient();' > test.ts`
-  `npx tsc test.ts` (look for constructor error)
-  No error with `@pnp/sp@2.0.13`

**Comment by marwankhalili** [comment]
Should note that I'm currently using v2.2.0 without a constructor parameter and it appears to be working just as before. 
```typescript
// @ts-ignore
const spClient = new SPHttpClient();
```

**Comment by patrick-rodgers** [comment]
Thanks for reporting this @marwankhalili we def needed to clean those constructors up. Should be all fixed now in 2.3.0 scheduled to go out today, March 12, 2021.
